docs/docs/fundamentals/gesture-composition.md

```diff
@@ -200,6 +200,209 @@ function App() {
+  },
+  outer: {
+    width: 250,
+    height: 250,
+    backgroundColor: 'lightblue',
+  },
+  inner: {
+    width: 100,
+    height: 100,
+    backgroundColor: 'blue',
+    alignSelf: 'center',
+  },
+});
+```
+
+## requiredToFailByExternalGesture
+
+`requiredToFailByExternalGesture` works similarily to `requireExternalGestureToFail` but the direction of the relation is reversed - insted of making the gesture it's called on wait for failure of gestures passed as arguments, it's making gestures passed as arguments wait for the gesture it's called on. It's especially usefull for making lists where the `ScrollView` component needs to wait for every gesture underneath it. All that's required is passing a ref of the `ScrollView` to the gesture object, for example:
+
+```jsx
+import React, { useRef } from 'react';
+import { StyleSheet } from 'react-native';
+import {
+  GestureDetector,
+  Gesture,
+  GestureHandlerRootView,
+  ScrollView,
+} from 'react-native-gesture-handler';
+import Animated, {
+  useSharedValue,
+  useAnimatedStyle,
+  withTiming,
+} from 'react-native-reanimated';
+
+const ITEMS = ['red', 'green', 'blue', 'yellow'];
+
+function Item({ backgroundColor, scrollRef }) {
+  const scale = useSharedValue(1);
+  const zIndex = useSharedValue(1);
+
+  const pinch = Gesture.Pinch()
+    .requiredToFailByExternalGesture(scrollRef)
+    .onBegin(() => {
+      zIndex.value = 100;
+    })
+    .onChange((e) => {
+      scale.value *= e.scaleChange;
+    })
+    .onFinalize(() => {
+      scale.value = withTiming(1, undefined, (finished) => {
+        if (finished) {
+          zIndex.value = 1;
+        }
+      });
+    });
+
+  const animatedStyles = useAnimatedStyle(() => ({
+    transform: [{ scale: scale.value }],
+    zIndex: zIndex.value,
+  }));
+
+  return (
+    <GestureDetector gesture={pinch}>
+      <Animated.View
+        style={[
+          { backgroundColor: backgroundColor },
+          styles.item,
+          animatedStyles,
+        ]}
+      />
+    </GestureDetector>
+  );
+}
+
+export default function Example() {
+  const scrollRef = useRef();
+
+  return (
+    <GestureHandlerRootView style={styles.container}>
+      <ScrollView style={styles.container} ref={scrollRef}>
+        {ITEMS.map((item) => (
+          <Item backgroundColor={item} key={item} scrollRef={scrollRef} />
+        ))}
+      </ScrollView>
+    </GestureHandlerRootView>
+  );
+}
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+  },
+  item: {
+    flex: 1,
+    aspectRatio: 1,
+  },
+});
+```
+
+## simultaneousWithExternalGesture
+
+`simultaneousWithExternalGesture` allows gestures across different components to be recognized simultaneously. We can modify the example from `requireExternalGestureToFail` to showcase this: let's say you have two nested views, again both with tap gesture attached. This time, both of them require one tap, but tapping the inner one should also activate the gesture attached to the outer view:
```

Comment:
People rarely read documentation from start to finish. They mostly skim the headings and code so refering to previous sections isn't that useful

It's better to repeat the full intent of the paragraph rather than refering to previous sections